lib/src/values/random_logic_value.dart

```diff
@@ -50,28 +59,31 @@ extension RandLogicValue on math.Random {
     } else {
       if (width <= LogicValue._INT_BITS) {
         LogicValue ranNum;
-        if (width < 32) {
+        if (width <= 32) {
           ranNum = LogicValue.ofInt(nextInt(1 << width), width);
         } else {
           ranNum = LogicValue.ofInt(_nextBigInt(numBits: width).toInt(), width);
         }
 
-        return LogicValue.ofInt(
-            ranNum.toInt() % (max == null ? (1 << width) - 1 : max.toInt()),
-            width);
+        if (max == null) {
+          return ranNum;
+        } else {
+          if (max is BigInt) {
+            throw InvalidRandomLogicValueException(
```

Comment:
I don't think we have to throw an exception here.  Just because it's a bigint doesn't mean we cant use it, right?